### PR TITLE
Feature/cid 2731/establish ws connection on startup

### PR DESCRIPTION
**Pull request by geoandri**
## 🛠 Changes made
Please explain here the changes you made in this PR.

## ✨ Type of change
Please delete the options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## 🧪 How Has This Been Tested?
Please describe the tests that you ran to verify your changes.

- [x] Unit tests

## 🏎 Checklist:
- [x] My code follows the style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have made corresponding changes to the documentation (README.md)
- [x] My commit message clearly reflects the changes made
- [x] Assigned the appropriate labels (version, PR type, etc.)

### CODE REVIEW

File: src/main/kotlin/net/leanix/githubagent/config/WebSocketClientConfig.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    private val gitHubEnterpriseProperties: GitHubEnterpriseProperties
+) {
+
+    @Retry(name = "ws-init-session")
+    fun initSession(): StompSession {
+        val headers = WebSocketHttpHeaders()
+        val stompHeaders = StompHeaders()
+        stompHeaders["Authorization"] = "Bearer ${authService.getBearerToken()}"
+        stompHeaders["GitHub-Enterprise-URL"] = gitHubEnterpriseProperties.baseUrl
+        stompHeaders["GiHub-Enterprise-App-Id"] = gitHubEnterpriseProperties.gitHubAppId
```
Comment (henriqamaral) [inline src/main/kotlin/net/leanix/githubagent/config/WebSocketClientConfig.kt:34]: I think we need to send the agent version here, not the app id

File: src/main/kotlin/net/leanix/githubagent/client/AuthClient.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.http.MediaType.APPLICATION_FORM_URLENCODED_VALUE
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestHeader
+
+@FeignClient(
+    name = "authentication",
+    url = "\${leanix.auth.access-token-uri}",
+)
+fun interface AuthClient {
```
[the remark this answers is not in the document]
Comment (geoandri) [reply src/main/kotlin/net/leanix/githubagent/client/AuthClient.kt:15]: I think `LeanixAuthClient` is more appropriate.

File: src/main/kotlin/net/leanix/githubagent/config/WebSocketClientConfig.kt (hunk 1 above)
Comment (geoandri) [reply src/main/kotlin/net/leanix/githubagent/config/WebSocketClientConfig.kt:34]: @henriqamaral when you say the agent version, you are referring to the one that is part of the gradle file?

Comment (henriqamaral) [reply src/main/kotlin/net/leanix/githubagent/config/WebSocketClientConfig.kt:34]: Hi @geoandri,
Yes, like we had on the old agents(brokers)

File: src/main/kotlin/net/leanix/githubagent/handler/BrokerStompSessionHandler.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.stereotype.Component
+
+@Component
+class BrokerStompSessionHandler : StompSessionHandlerAdapter() {
+
+    private val logger = LoggerFactory.getLogger(BrokerStompSessionHandler::class.java)
+
+    override fun afterConnected(session: StompSession, connectedHeaders: StompHeaders) {
+        logger.info("connected to the server: ${session.sessionId}")
+        session.subscribe("/user/queue/repositories-string", this)
```
[the remark this answers is not in the document]
Comment (geoandri) [reply src/main/kotlin/net/leanix/githubagent/handler/BrokerStompSessionHandler.kt:17]: Nothing has been decided yet on this. It is just the example from PoC. @henriqamaral what should be the values here, or we can just remove this until it is decided.

[the remark this answers is not in the document]
Comment (henriqamaral) [reply src/main/kotlin/net/leanix/githubagent/handler/BrokerStompSessionHandler.kt:17]: we can just remove for now

File: src/main/kotlin/net/leanix/githubagent/handler/BrokerStompSessionHandler.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        payload: ByteArray,
+        exception: Throwable,
+    ) {
+        logger.error("handleException", exception)
+        logger.error(exception.message)
+    }
+
+    override fun handleTransportError(session: StompSession, exception: Throwable) {
+        logger.error("handleTransportError", exception)
+        logger.error(exception.message)
```
[the remark this answers is not in the document]
Comment (geoandri) [reply src/main/kotlin/net/leanix/githubagent/handler/BrokerStompSessionHandler.kt]: I have added the same approach that is present in the PoC in the latest.

**Review by henriqamaral — APPROVED** [review]

**Pull request merged**